p256/src/arithmetic/field/field32.rs

```diff
@@ -0,0 +1,252 @@
+//! 32-bit secp256r1 field element algorithms.
+
+use super::MODULUS;
+use crate::arithmetic::util::*;
+use elliptic_curve::bigint::{U256, U512};
+
+pub(super) const fn add(a: U256, b: U256) -> U256 {
+    let a = a.as_words();
+    let b = b.as_words();
+
+    // Bit 256 of p is set, so addition can result in nine words.
+    let (w0, carry) = adc(a[0], b[0], 0);
```

Comment:
Note that if you used `as_limbs`, they define a word size specific `Limb::adc` operation which could be used here instead (and get rid of `util.rs` in the process), however that's a refactoring I can potentially do as a followup.